### ISSUE TITLE
Matcher: Quite many "OOPS" pages when "additional grok patterns" field

### ISSUE THREAD

**Issue by stoerr** [stoerr/GrokConstructor#4]
There are quite a lot misusages of that field. Unclear: do people not understand what the field is for, or accidentially use the wrong field? If it's accidential, we can hide it per Javascript, so people deliberately have to make it visible. (New checkbox in the page.)

**Comment by stoerr** [comment]
Unclear whether anybody actually uses this. We'll ensure that the string grokadditional only occurs in the log when somebody uses this, and look after a while.

**Comment by brianshen1990** [comment]
When I compiled it and used in windows, it have the same problem.

**Comment by stoerr** [comment]
@brianshen1990 Have you identified the reason for these? If not please include the content of one or two error pages that I can guess.

**Comment by stoerr** [comment]
https://github.com/stoerr/GrokConstructor/issues/33 seems a duplicate of this.

**Comment by stoerr** [comment]
Now that was quite an interesting bug! I just checked the continuation against the rest of the line, thinking if it would match this, it would also match in the context of the full line. That isn't true, though, since some patterns (e.g. WORD) start with a word boundary marker, which always matches at the start of the rest of the line, but not necessarily in the context of the full line. So I've introduced an additional check to weed that out.
